controllers/datadogagent/kubestatemetrics.go

```diff
@@ -0,0 +1,194 @@
+package datadogagent
+
+import (
+	"context"
+
+	datadoghqv1alpha1 "github.com/DataDog/datadog-operator/api/v1alpha1"
+	"github.com/DataDog/datadog-operator/pkg/controller/utils/datadog"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	kubeStateMetricsRBACName = "kube-state-metrics-core"
+	defaultKSMCoreConfigMap  = `
+---
+cluster_check: true
+init_config:
+instances:
+  - collectors:
+      - pods
+      - replicationcontrollers
+      - statefulsets
+      - nodes
+      - cronjobs
+      - jobs
+      - replicasets
+    telemetry: true
+  - collectors:
+      - deployments
+      - configmaps
+      - services
+      - endpoints
+      - daemonsets
+      - horizontalpodautoscalers
+      - limitranges
+      - resourcequotas
+      - secrets
+      - namespaces
+      - persistentvolumeclaims
+      - persistentvolumes
+    telemetry: true
```

Comment:
should we have once instance by collector, as we discussed?
Or maybe for now, only one instance?

Reply:
moved to 1 instance